Tutorials/GraphHamiltonian/J1J2/j1j2.py

```diff
@@ -0,0 +1,114 @@
+#Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+
+from __future__ import print_function
+import json
+import numpy as np
+import networkx as nx
+
+#Sigma^z*Sigma^z interactions
+sigmaz = [[1, 0], [0, -1]]
+mszsz = (np.kron(sigmaz, sigmaz))
+
+#Exchange interactions
+exchange = np.asarray([[0, 0, 0, 0], [0, 0, 2, 0], [0, 2, 0, 0], [0, 0, 0, 0]])
+
+#Couplings J1 and J2
+J = [1, 0.4]
+L = 20
+
+pars = {}
+
+# Define bond operators, labels, and couplings
+bond_operator = [
+    (J[0] * mszsz).tolist(),
+    (J[1] * mszsz).tolist(),
+    (-J[0] * exchange).tolist(),
+    (J[1] * exchange).tolist(),
+]
+
+bond_color = [1, 2, 1, 2]
+
+# Define custom graph
+G = nx.Graph()
+for i in range(L):
+    G.add_edge(i, (i + 1) % L, color=1)
+    G.add_edge(i, (i + 2) % L, color=2)
+
+    G.add_edge((i + 1) % L, i, color=1)
+    G.add_edge((i + 2) % L, i, color=2)
+
+edge_colors = [[u, v, G[u][v]['color']] for u, v in G.edges]
+
+# Specify custom graph
+pars['Graph'] = {
+    'Edges': list(G.edges),
+    'EdgeColors': edge_colors,
+}
+
+#We chose a spin 1/2 hilbert space with total Sigmaz=0
+pars['Hilbert'] = {
+    'Name': 'Spin',
+    'S': 0.5,
+    'TotalSz': 0,
+    'Nspins': L,
+}
+
+#defining our custom hamiltonian
+pars['Hamiltonian'] = {
+    'Name': 'Graph',
+    'SiteOps': [],
```

Comment:
Maybe we should just check that at least one among `SiteOps` and `BondOps` is defined in the constructor of  `GraphHamiltonian`. In this way, we can avoid to explicitly write 

````python
'SiteOps': [],
````

or similarly when there are only `SiteOps` but no `BondOps` (boring hamiltonian in that case though!)

Reply:
Great point, I added some checks in `GraphHamiltonian` that create empty arrays for SiteOps or BondOps if they aren't specified (and check to make sure that at least one is specified). I've also added the behavior where users aren't required to specify `BondOpColors` if they are all the same color.